src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -688,6 +697,20 @@ private boolean isBlocked(InetAddress address) {
         return false;
     }
 
+    public static boolean matchesHostPatterns(List<String> hostPatterns, InetAddress address) {
+        WildcardMatcher hostMatcher = WildcardMatcher.NONE;
+        if (hostPatterns != null && !hostPatterns.isEmpty()) {
+            hostMatcher = WildcardMatcher.from(hostPatterns);
+        }
```

Comment:
Since this will be checked per request, can we have this pre-computed in today's implementation or do you think that requires restructuring more pieces of the plugin?

Reply:
I pushed a commit to refactor and only compute this once, but I still need to test the dynamic update scenario where an administrator updates the auth failure listener dynamically to change this field.